### PR TITLE
Removing a lot of deprecated calls - mostly in Factories.

### PR DESCRIPTION
**Pull request by rarog**

### CODE REVIEW

File: src/BjyAuthorize/Provider/Role/ObjectRepositoryProvider.php [hunk 1]
```diff
@@ -10,23 +10,23 @@
 
 use BjyAuthorize\Acl\HierarchicalRoleInterface;
 use BjyAuthorize\Acl\Role;
-use Doctrine\Common\Persistence\ObjectRepository;
+use Doctrine\Persistence\ObjectRepository;
```
Comment (kokspflanze) [inline src/BjyAuthorize/Provider/Role/ObjectRepositoryProvider.php:13]: so we need also a new dev dependency
`doctrine/persistence: ^1.3.3 || ^2.0`

Comment (rarog) [reply src/BjyAuthorize/Provider/Role/ObjectRepositoryProvider.php:13]: You are right, I replaced the dependency to be more specific.

Comment (kokspflanze) [reply src/BjyAuthorize/Provider/Role/ObjectRepositoryProvider.php:13]: i think the composer.lock, need the update too

Comment (rarog) [reply src/BjyAuthorize/Provider/Role/ObjectRepositoryProvider.php:13]: Oh, I actually deleted it, as I thought vthat the commit was accidental, as for libraries it's usually of no effect as per https://getcomposer.org/doc/02-libraries.md#lock-file

I can restore and update it.

**Pull request merged**